### PR TITLE
PETSc examples: Update DMPlexCreateSphereMesh() and DMAddBoundary() calls after changes in PETSc's API

### PR DESCRIPTION
**Pull request by valeriabarra**
I have updated the couple of instances where we were calling `DMPlexCreateSphereMesh` after changes in PETSc.

### CODE REVIEW

**Comment by valeriabarra** [comment]
Need to update `junit`

File: examples/petsc/area.c [hunk 1]
```diff
@@ -116,7 +116,7 @@ int main(int argc, char **argv) {
     CHKERRQ(ierr);
   } else {
     // Create the mesh as a 0-refined sphere. This will create a cubic surface, not a box
-    ierr = DMPlexCreateSphereMesh(PETSC_COMM_WORLD, topodim, simplex, &dm);
+    ierr = DMPlexCreateSphereMesh(PETSC_COMM_WORLD, topodim, simplex, 1., &dm);
```
Comment (jeremylt) [inline examples/petsc/area.c:119]: Question - do we want to update this with a macro, like we did for DMPlexGetClosureIndices?

**Comment by jedbrown** [comment]
Have we tested it with 3.13?  If it works at all there, then we could use the compatibility macro, otherwise we can remove them and say the example is for >=3.14.

**Comment by valeriabarra** [comment]
> Have we tested it with 3.13? If it works at all there, then we could use the compatibility macro, otherwise we can remove them and say the example is for >=3.14.

I have just switched to 3.13 to test this and tried to simply reconfigure and compile as usual with 
```
./$PETSC_ARCH/lib/petsc/conf/reconfigure-$PETSC_ARCH.py && make all
```
but now I'm fighting a configure error I'm getting with `MUMPS`

**Comment by jedbrown** [comment]
You don't need MUMPS at all for these tests.  I would use a separate PETSC_ARCH so you don't have to reconfigure.

File: examples/petsc/area.c (hunk 1 above)
Comment (valeriabarra) [reply examples/petsc/area.c:119]: Added them. Thanks

**Comment by valeriabarra** [comment]
Done

**Comment by jeremylt** [comment]
If I recall correctly, Azure is failing because the version checking macros don't work when you're not on a released version (which the PETSc build for Azure isn't).

Other than that, this seems good to me.

**Comment by valeriabarra** [comment]
Thanks Jeremy. Can I go ahead with the merge?

**Comment by jedbrown** [comment]
What's going on here? It's a fresh image.

https://dev.azure.com/CEED-ECP/libCEED/_build/results?buildId=3014&view=logs&j=6b902995-b73d-5f5c-66fd-a7f66c857d2c&t=a1c457be-0fbe-5525-b6b0-ac3cf58f8d6c

**Comment by jedbrown** [comment]
DMAddBoundary isn't in this PR?

**Comment by valeriabarra** [comment]
In [beae6a5](https://github.com/CEED/libCEED/pull/600/commits/beae6a5ce4cd87fc6ef6384d5f79ffdc2f341d81) I updated the call to `DMAddBoundary()` and in [23469ac](https://github.com/CEED/libCEED/pull/600/commits/23469acc68d9f2a98f561c7b5de6ee2a3745201a) I aded the relative compatibility macros. I tested with both 3.13 and master on my local machine and it was fine.

**Comment by jeremylt** [comment]
I think the fluids example also needs the new compatibility macro (and maybe solids too iirc?)

**Comment by valeriabarra** [comment]
> I think the fluids example also needs the new compatibility macro (and maybe solids too iirc?)

Thanks for the reminder Jeremy! Update those, too.

`prove-all` passes clean locally for me.

**Comment by jedbrown** [comment]
It passed on Azure too.

**Comment by jeremylt** [comment]
Travis needs the new commit hash for PETSc to be able to pass.

**Comment by valeriabarra** [comment]
Should I just update [this line](https://github.com/CEED/libCEED/blob/9f2752f144770f23a090e84e243e59bcab3f0c80/.travis.yml#L153) with the latest hash?

**Comment by jeremylt** [comment]
Yes, that should be all that's needed

**Pull request merged**